renderer/brave_wallet/brave_wallet_render_frame_observer.cc

```diff
@@ -44,10 +45,8 @@ bool BraveWalletRenderFrameObserver::CanCreateProvider() {
     return false;
   }
 
-  // Wallet provider objects won't be generated for third party iframe
-  if ((!render_frame()->IsMainFrame() &&
-       render_frame()->GetWebFrame()->IsCrossOriginToOutermostMainFrame()) ||
-      !render_frame()->GetWebFrame()->GetDocument().IsSecureContext()) {
+  // Wallet provider objects should only be created in secure contexts
+  if (!render_frame()->GetWebFrame()->GetDocument().IsSecureContext()) {
```

Comment:
Is this case already tested?

Reply:
Are we guarding/testing against these case 
- `Top level http://a.com/ with <iframe src=”[https://a.com](http://a.com/)”> -> blocked (insecure/3p)`
- `Top level [https://b.com](http://a.com/) with <iframe src=”http://a.com/”> with <iframe src=”[https://b.com](http://a.com/)”> -> blocked (insecure)`
- `Top level [https://a.com](http://a.com/) with <iframe src=”[https://a.com](http://a.com/)” sandbox> -> blocked (sandbox)`
- `Top level data://foo with <iframe src=”data://bar”> -> blocked (insecure)`
- `Top level file://foo with <iframe src=”file://bar”> -> blocked (3p)`

Reply:
For the other ones rebased the missing ones (top level insecure) here:
https://github.com/brave/brave-core/pull/13783/commits/f6dae3088b071f5a5e8067238167b7a0a87e6182